dataland-keycloak-adapter/src/main/kotlin/org/dataland/keycloakAdapter/auth/DatalandApiKeyAuthentication.kt

```diff
@@ -0,0 +1,24 @@
+package org.dataland.keycloakAdapter.auth
+
+import org.dataland.datalandapikeymanager.openApiClient.model.ApiKeyMetaInfo
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+
+/**
+ * This authentication class represents a successful authentication via a dataland API-Key
+ */
+class DatalandApiKeyAuthentication(
+    private val apiKey: String,
+    val apiKeyMetaInformation: ApiKeyMetaInfo,
+) : DatalandAuthentication() {
+    override val userId: String
+        get() = apiKeyMetaInformation.keycloakUserId!!
+
+    override fun getAuthorities(): List<GrantedAuthority> {
+        return apiKeyMetaInformation.keycloakRoles?.map { SimpleGrantedAuthority(it) } ?: emptyList()
```

Comment:
I see that we store these roles alongside with the api key, but can you explain to me why?